buildchain/buildchain/vagrant.py

```diff
@@ -40,16 +42,53 @@ def mkdir_dot_vagrant() -> None:
         'uptodate': [True],
     }
 
+def task_vagrantupnoprov() -> types.TaskDict:
+    """Run `vagrant up` without provisioning a development environment."""
+    vagrantupnoprov = [config.VAGRANT, 'up']
+    vagrantupnoprov.extend(map(shlex.quote,
+        config.VAGRANT_UP_ARGS + tuple(('--no-provision',))))
+    return {
+        'actions': [doit.tools.LongRunning(' '.join(vagrantupnoprov))],
+        'file_dep': [constants.VAGRANT_SSH_KEY_PAIR],
+        'task_dep': [],
+        'uptodate': [False],
+    }
+
+def task_vagrantsnapshot() -> types.TaskDict:
+    """Snapshot development environment."""
+    vagrantsnap = [config.VAGRANT, 'snapshot', 'save', '--force', 'bootstrap', config.VAGRANT_SNAPSHOT_NAME]
+
+    return {
+        'actions': [doit.tools.LongRunning(' '.join(vagrantsnap))],
+        'file_dep': [],
+        'task_dep': ['vagrantupnoprov'],
+        'uptodate': [run_once]
+    }
+
+def task_vagrantreset() -> types.TaskDict:
```

Comment:
When we call `vagrant up` the task is called `vagrantup`: I would keep this logic and call this task `vagrantrestore` (or `vagrant_restore` for the sake of legibility).

Reply:
If we call this task `vagrant_restore`, then `vagrantup` should be `vagrant_up` (for consistency)...

Reply:
Yup, indeed.

You can either do it here or create a debt issue to do the rename in a dedicated PR.

Reply:
Done